src/type/definition.js

```diff
@@ -82,11 +85,16 @@ export function isInputType(type: ?GraphQLType): boolean %checks {
   );
 }
 
-export function assertInputType(type: ?GraphQLType): GraphQLInputType {
-  invariant(
-    isInputType(type),
-    `Expected ${String(type)} to be a GraphQL input type.`
-  );
+export function assertInputType(
+  type: ?GraphQLType,
+  typeNode?: ASTNode,
+  source?: Source
+): GraphQLInputType {
+  invariant(isInputType(type),
+    validationError(
+      source,
+      typeNode,
+      `Expected ${String(type)} to be a GraphQL input type.`));
   return type;
```

Comment:
These methods are used throughout and not only in validation, so instead anywhere doing schema validation should be using `isInputType` and throwing the appropriate error instead of this general purpose function changing